app/models/api/order_detail.rb

```diff
@@ -0,0 +1,35 @@
+class Api::OrderDetail
+  def initialize(order_detail)
+    @order_detail = order_detail
+  end
+
+  def to_h
+    { account: account_to_hash, ordered_for: ordered_for_to_hash }
+  end
+
+  private
+
+  def account
+    @account = @order_detail.try(:account) unless defined?(@account)
```

Comment:
Since we're doing the 404 catching in the controller, do we need to handle the nil case? Also, could this just be `@account ||= @order_detail.account` as opposed to the `defined?`?